string[] names
uint16[] x
uint16[] y
float32[] theta
